src/main/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/services/OcrDataParser.java

```diff
@@ -0,0 +1,63 @@
+package uk.gov.hmcts.reform.bulkscanccdeventhandler.services;
+
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.bulkscanccdeventhandler.model.CcdCollectionElement;
+import uk.gov.hmcts.reform.bulkscanccdeventhandler.model.OcrData;
+import uk.gov.hmcts.reform.bulkscanccdeventhandler.model.OcrDataField;
+import uk.gov.hmcts.reform.bulkscanccdeventhandler.model.ResultOrErrors;
+
+import java.util.List;
+import java.util.Map;
+
+import static uk.gov.hmcts.reform.bulkscanccdeventhandler.model.ExceptionRecordFieldNames.SCAN_OCR_DATA;
+import static uk.gov.hmcts.reform.bulkscanccdeventhandler.model.ResultOrErrors.errors;
+import static uk.gov.hmcts.reform.bulkscanccdeventhandler.model.ResultOrErrors.result;
+
+/**
+ * Converts raw exception record data representing OCR data fields
+ * (in the form of a list of maps of objects) into a strongly typed object.
+ */
+@Service
+public class OcrDataParser {
+
+    private final ObjectMapper mapper;
+
+    public OcrDataParser(ObjectMapper mapper) {
+        this.mapper = mapper;
+    }
+
+    public ResultOrErrors<OcrData> parseOcrData(
```

Comment:
Curious of the bigger picture... perhaps this class could just throw an exception if json can't be mapped to expected model??

Reply:
after all, the ocr data json not being in the correct format is not a normal scenario.

Reply:
We retrieve multiple instances of this class, whenever we want to retrieve simple or complex structures from the exception record data in order to create case data. Because we shouldn't stop at the first error but instead give a complete list of errors, throwing an exception wouldn't be best for this.

Reply:
> after all, the ocr data json not being in the correct format is not a normal scenario.

Yes, but there are also scanned documents, not related with OCR data, which can cause errors, too.

Reply:
ok, it will be easier to see later with more usage I guess

Reply:
hopefully we're not overengineering this **sample** app

Reply:
My intention is to go as simple as possible, but there are limits. Our sample app must make Bulk Scanning cases and if it can't, it should return a list of reasons. There can be many errors even within OCR data with valid format, as its fields can be mapped to various types.